sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -233,4 +238,185 @@ export class RestrictionController extends BaseController {
+   * @param institutionId
+   * @returns Institution Restrictions.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/institution/:institutionId")
+  async getInstitutionRestrictions(
+    @Param("institutionId") institutionId: number,
+  ): Promise<RestrictionSummaryDTO[]> {
+    const institutionRestrictions =
+      await this.institutionRestrictionService.getInstitutionRestrictionsById(
+        institutionId,
+      );
+    return institutionRestrictions?.map((institutionRestriction) => ({
+      restrictionId: institutionRestriction.id,
+      restrictionType: institutionRestriction.restriction.restrictionType,
+      restrictionCategory:
+        institutionRestriction.restriction.restrictionCategory,
+      description: institutionRestriction.restriction.description,
+      createdAt: institutionRestriction.createdAt,
+      updatedAt: institutionRestriction.updatedAt,
+      isActive: institutionRestriction.isActive,
+    }));
+  }
+
+  /**
+   * Rest API to get the details for view institution restriction.
+   * @param institutionId
+   * @param institutionRestrictionId
+   * @returns Institution restriction detail view.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get(
+    "/institution/:institutionId/institutionRestriction/:institutionRestrictionId",
+  )
+  async getInstitutionRestrictionDetail(
+    @Param("institutionId") institutionId: number,
+    @Param("institutionRestrictionId") institutionRestrictionId: number,
+  ): Promise<RestrictionDetailDTO> {
+    const institutionRestriction =
+      await this.institutionRestrictionService.getInstitutionRestrictionDetailsById(
+        institutionId,
+        institutionRestrictionId,
+      );
+    if (!institutionRestriction) {
+      throw new NotFoundException(
+        "The institution restriction does not exist.",
+      );
+    }
+    return {
+      restrictionId: institutionRestriction.id,
+      restrictionType: institutionRestriction.restriction.restrictionType,
+      restrictionCategory:
+        institutionRestriction.restriction.restrictionCategory,
+      description: institutionRestriction.restriction.description,
+      createdAt: institutionRestriction.createdAt,
+      updatedAt: institutionRestriction.updatedAt,
+      createdBy: institutionRestriction.creator
+        ? `${institutionRestriction.creator.lastName}, ${institutionRestriction.creator.firstName}`
+        : "",
+      updatedBy: institutionRestriction.modifier
+        ? `${institutionRestriction.modifier.lastName}, ${institutionRestriction.modifier.firstName}`
+        : "",
+      isActive: institutionRestriction.isActive,
+      restrictionNote: institutionRestriction.restrictionNote?.description,
+      resolutionNote: institutionRestriction.resolutionNote?.description,
+    };
+  }
+
+  /**
+   * Rest API to add a new provincial restriction to Institution.
+   * @param userToken
```

Comment:
can you update the comment regarding the federal restriction

Reply:
The federal restriction comment doesn't apply to Institution because only provincial restriction of type Designation is applicable for institution. Federal restrictions are out of it's scope.

Reply:
ya, you can mention this is comments for future reference

Reply:
comment added.